components/camel-test/camel-test-junit5/src/main/java/org/apache/camel/test/junit5/CamelTestSupport.java

```diff
@@ -144,15 +145,16 @@ public void beforeAll(ExtensionContext context) {
             testConfigurationBuilder.withCreateCamelContextPerClass(perClassPresent);
             contextManager = contextManagerFactory.createContextManager(ContextManagerFactory.Type.BEFORE_ALL,
                     testConfigurationBuilder, camelContextConfiguration);
+            ExtensionContext.Store globalStore = context.getStore(ExtensionContext.Namespace.GLOBAL);
+            contextManager.setGlobalStore(globalStore);
         }
-
-        ExtensionContext.Store globalStore = context.getStore(ExtensionContext.Namespace.GLOBAL);
-        contextManager.setGlobalStore(globalStore);
     }
 
     @Override
     public void afterAll(ExtensionContext context) {
-        contextManager.stop();
+        if(Objects.nonNull(contextManager)) {
+            contextManager.stop();
+        }
```

Comment:
You can ignore if the contextManager is null here. So, just add a check for nullity and, if present, then stop. 

There may be cases when the after* hooks could be called multiple times and/or without the contextManager having been created (I'm thinking when using `@Nested` tests, but we'll see after I run a full build).

Reply:
not sure I get your request, the non-null check is there?

Reply:
For instance:
```
 if (contextManager != null) {
            // It may be null in some occasion, such as when failing to initialize the context
            contextManager.stop();
        }
```